pennylane_lightning_gpu/src/util/DataBuffer.cu

```diff
@@ -0,0 +1,37 @@
+//#pragma once
+#include "DataBuffer.hpp"
+
+namespace Pennylane::CUDA {
+
+template void DataBuffer<float2, int>::setElements(int &num_indices,
+                                                   float2 *value, int *indices,
+                                                   bool async);
+template void DataBuffer<double2, int>::setElements(long &num_indices,
+                                                    double2 *value,
+                                                    long *indices, bool async);
+
+template <class DeviceDataT, class index_type>
+__global__ void cuda_element_set(index_type num_indices, DeviceDataT *value,
+                                 index_type *indices,
+                                 DeviceDataT *gpu_buffer_) {
+    const unsigned int i = blockIdx.x * blockDim.x + threadIdx.x;
+    if (i < num_indices) {
+        gpu_buffer_[indices[i]] = value[i];
+    }
+}
+
+template <class GPUDataT, class DevTagT>
+template <class DeviceDataT, class index_type>
```

Comment:
Can we combine these templates into a single call?

Reply:
let me check!

Reply:
Tried to combine these templates into a single call but failed. Will keep as it is.

Reply:
I'll look into this.

Reply:
> I'll look into this.

[Here](https://stackoverflow.com/questions/58880487/template-method-within-template-class-definition-does-not-match-declaration) is something that might be useful.